dom/events/scrolling/wheel-event-groups-target-removal.html

```diff
@@ -0,0 +1,94 @@
+<!DOCTYPE HTML>
+<html>
+<head>
+<meta name="viewport" content="width=device-width,initial-scale=1,minimum-scale=1">
+<meta name="timeout" content="long">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-actions.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script src="scroll_support.js"></script>
+<style>
+body {
+  margin: 0;
+  padding: 0;
+  height: 200vh;
+}
+
+.spacer {
+  width: 100%;
+  height: 25px;
+  padding: 0;
+  margin: 0;
+}
+
+</style>
+<head>
+<body onload=runTest()>
+<div id="initialTarget" class="spacer" style="background: red"></div>
+<div id="firstRootSpacer" class="spacer" style="background: green"></div>
+<div id="secondRootSpacer" class="spacer" style="background: blue"></div>
+</body>
+
+<script>
+
+function runTest() {
+  promise_test(async (t) => {
+    await waitForCompositorCommit();
+
+    // Wait some extra time to ensure nothing from prior tests impacts this one.
+    // At a minimum this should be greater than 500ms to ensure that the wheel events
+    // in this test are in a new wheel event group
+    await new Promise(resolve => step_timeout(resolve, 1000));
+
+    await waitForCompositorCommit();
+
+    let initialTarget = document.getElementById('initialTarget');
+
+    let wheelEventTargets = [];
+    let removedInitialTarget = false;
+
+    // Remove the initial element the wheel event is targetted at once we fire the
+    // first wheel event, then log the subsequent wheel event targets.
+    function removeInitialElement(e) {
+      wheelEventTargets.push(e.target);
+      if (!removedInitialTarget) {
+        removedInitialTarget = true;
+        e.target.remove();
+      }
+    }
+    window.addEventListener("wheel", removeInitialElement, {passive: true});
+
+    await waitForCompositorCommit();
+
+    await new test_driver.Actions()
+      .addWheel("wheel1")
+      .scroll(40, 2, 0, 30, {origin: "viewport"})
+      .pause(1)
+      .scroll(40, 2, 0, 30, {origin: "viewport"})
+      .pause(1)
+      .scroll(40, 2, 0, 30, {origin: "viewport"})
+      .send();
+
+    // TODO(dlrobertson): Use the scrollend event here to wait for the
+    // wheel scroll to finish instead of waitForAnimationEnd().
+    await waitForAnimationEnd(() => { return document.scrollingElement.scrollTop; });
+    await waitForCompositorCommit();
+
+    document.scrollingElement.scrollTo({top: 0, left: 0, behavior: "instant"});
+
+    // The first wheel event should be targetted at the removed element.
+    assert_equals(wheelEventTargets.shift(), initialTarget,
+                  "Initial wheel event is has the removed element as the target");
+
+    wheelEventTargets.forEach((wheelEventTarget) => {
+      assert_not_equals(wheelEventTarget, initialTarget,
+                        "All following wheel events are not targetting the removed element");
+    });
```

Comment:
Note that browser engines do different things in the event target mutation test cases. E.g. Gecko and WebKit will essentially fallback to the non wheel event groups event target after the original event target is mutated in some way, while Blink will only retarget once. I greatly weakened this assertion, so that all engines should pass. We might want to circle back at some point once there is some sort of consensus.